contracts/actions/curveusd/advanced/CurveUsdSwapper.sol

```diff
@@ -225,30 +233,43 @@ contract CurveUsdSwapper is CurveUsdHelper, ExchangeHelper, GasFeeHelper, AdminA
         swapRoutes.pools[4] = swapZapPools[4];
     }
 
-    function takeSwapAndGasCostFee(
-        uint256 _amount,
+    function takeSwapFee(
+        uint256 _sellAmount,
         address _user,
         address _token,
-        uint256 _dfsFeeDivider,
-        uint32 _gasUsed
-    ) internal returns (uint256 feeAmount) {
-        if (_dfsFeeDivider != 0 && Discount(DISCOUNT_ADDRESS).isCustomFeeSet(_user)) {
-            _dfsFeeDivider = Discount(DISCOUNT_ADDRESS).getCustomServiceFee(_user);
-        }
-
-        // we need to take the fee for tx cost as well, as it"s in a strategy
-        if (_gasUsed != 0) {
-            feeAmount += calcGasCost(_gasUsed, _token, 0);
+        uint24 _dfsFeeDivider
+    ) internal returns (uint256 feeAmount, uint256 dfsFeeDivider) {
+        dfsFeeDivider = _dfsFeeDivider;
+        if (dfsFeeDivider != 0 && Discount(DISCOUNT_ADDRESS).isCustomFeeSet(_user)) {
+            dfsFeeDivider = Discount(DISCOUNT_ADDRESS).getCustomServiceFee(_user);
         }
 
         // take dfs fee if set, and add to feeAmount
-        if (_dfsFeeDivider != 0) {
-            feeAmount += _amount / _dfsFeeDivider;
+        if (dfsFeeDivider != 0) {
+            feeAmount += _sellAmount / _dfsFeeDivider;
         }
 
-        // fee can"t go over 10% of the whole amount
-        if (feeAmount > (_amount / 10)) {
-            feeAmount = _amount / 10;
+        address walletAddr = FeeRecipient(FEE_RECIPIENT_ADDRESS).getFeeAddr();
+        _token.withdrawTokens(walletAddr, feeAmount);
+    }
+
+    function takeAutomationFee(
+        uint256 _destTokenAmount,
+        address _token,
+        uint256 _gasUsed,
+        uint256 _dfsFeeDivider
+    ) internal returns (uint256 feeAmount) {
+        // we need to take the fee for tx cost as well, as it"s in a strategy
+        
+        feeAmount += calcGasCost(_gasUsed, _token, 0);
+        
+        // gas fee can't go over 20% of the whole amount
+        if (feeAmount > (_destTokenAmount / 5)) {
+            feeAmount = _destTokenAmount / 5;
+        }
+    
+        if (_dfsFeeDivider != 0) {
```

Comment:
if a user has a custom discount divider eq to 0 we wont take AUTOMATION_DFS_FEE here, is this expected or bug?

Reply:
expected behaviour, so we dont need to add more params to curve usd swapper